### PR TITLE
feat: adding the need for conventional commits to our CI

### PR DESCRIPTION
**Pull request by cmeesters**
This PR attempts (!) to provide two new CI features:

- enforcing conventional commits (see https://www.conventionalcommits.org )
- and automatic releases

As this is a first trial, some tinkering might be necessary until its working.

### CODE REVIEW

**Comment by mptrsen** [comment]
This is a draft PR and not ready for review/merge?

**Comment by mptrsen** [comment]
Or... omit autorelease and we create releases when we feel like it?

**Comment by cmeesters** [comment]
(not a high prio right now, feel free to work on it)

**Comment by cmeesters** [comment]
I _think_ it is good now, except that the build step is replaced in newer PRs - so I would like to change this on "ready for review" only after the more current PRs are merged. Then, however, we can produce releases repeatedly.

File: .github/workflows/release-please.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+on:
+  push:
+    branches:
+      - main
```
Comment (mptrsen) [inline .github/workflows/release-please.yml:4]: Um, nobody should push to main directly if I'm not mistaken. Does this work?

**Review by mptrsen — COMMENTED** [review]
see comments

File: .github/workflows/release-please.yml (hunk 1 above)
Comment (cmeesters) [reply .github/workflows/release-please.yml:4]: The idea is to create a new (minor) release when a branch is merged into `main`. That entails a push.

This way, tinkering on an unprotected branch allows for review and correction before a new release is triggered.

**Review by mptrsen — APPROVED** [review]

**Comment by cmeesters** [comment]
Hm, might not work? Or will work upon next merge? This is the drawback. We have to wait.

**Pull request merged**